source/extensions/access_loggers/file/config.cc

```diff
@@ -22,11 +23,21 @@ FileAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
   const auto& fal_config =
       MessageUtil::downcastAndValidate<const envoy::config::accesslog::v2::FileAccessLog&>(config);
   AccessLog::FormatterPtr formatter;
-  if (fal_config.format().empty()) {
-    formatter = AccessLog::AccessLogFormatUtils::defaultAccessLogFormatter();
+
+ if (fal_config.access_log_format_case() == envoy::config::accesslog::v2::FileAccessLog::kFormat
+    || fal_config.access_log_format_case() == envoy::config::accesslog::v2::FileAccessLog::ACCESS_LOG_FORMAT_NOT_SET) {
+      if (fal_config.format().empty()) {
+        formatter = AccessLog::AccessLogFormatUtils::defaultAccessLogFormatter();
+      } else {
+        formatter.reset(new AccessLog::FormatterImpl(fal_config.format()));
+      }
+  } else if (fal_config.access_log_format_case() == envoy::config::accesslog::v2::FileAccessLog::kJsonFormat) {
+    const auto& json_format_map = this->convert_json_format_to_map(fal_config.json_format());
+    formatter.reset(new AccessLog::JsonFormatterImpl(json_format_map));
   } else {
-    formatter.reset(new AccessLog::FormatterImpl(fal_config.format()));
+    throw EnvoyException("Invalid access_log format provided");
```

Comment:
would it make sense to include the provided format and list of supported formats in this error message?

Reply:
Agreed